packages/ses/src/immutable-array-buffer.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable class-methods-use-this */
+import {
+  defineProperties,
+  arrayBufferSlice,
+  toStringTagSymbol,
+  isView,
+  speciesSymbol,
+} from './commons.js';
+
+/**
+ * `ImmutableArrayBuffer` is intended to be a peer of `ArrayBuffer` and
+ * `SharedArrayBuffer`, but only with the non-mutating methods they have in
+ * common. We're adding this to ses as if it was already part of the
+ * language, and we consider this implementation to be a shim for an
+ * upcoming tc39 proposal.
+ *
+ * As a proposal it would take additional steps that would the shim does not:
+ * - to have `ImmutableArrayBuffer` be shared between
+ *   threads (in spec speak, "agent") in exactly the same way
+ *   `SharedArrayBuffer` is shared between agents. Unlike `SharedArrayBuffer`,
+ *   sharing an `ImmutableArrayBuffer` does not introduce any observable
+ *   concurrency. Unlike `ArrayBuffer`, sharing an `ImmutableArrayBuffer`
+ *   does not detach anything.
```

Comment:
FYI, only "transferred" `ArrayBuffer` become detached on the web. If not transferring, the AB is simply copied, which is for an immutable buffer is observably indistinguishable from sharing. Basically an `ImmutableArrayBuffer` would simply not be "transferable" on the web, but it could remain "cloneable".

Reply:
Ah, I did not realize that structured clone has a distinct knob for clone vs transfer. I'll clarify that the advantage is sharing immutable data *without copying*.

Reply:
I still view copying or not as an optimization that engines could do regardless. There is no guarantee that they would not copy a cloned immutable array buffer, and if the engine was implementing copy-on-write, it could avoid copying a cloned array buffer already.